### PR TITLE
chore: remove python 3.7 support

### PR DESCRIPTION
**Pull request by CaselIT**
Drop python 3.7. The main reason is the lack of typing.Protocol, complicating typing introduction

### CODE REVIEW

**Review by vytas7 — CHANGES_REQUESTED** [review]
Looks good, but we need to reword the notes in `docs/changes/4.0.0.rst`.

File: docs/changes/4.0.0.rst [hunk 1]
```diff
@@ -15,10 +15,9 @@ Changes to Supported Platforms
...
-- End-of-life Python 3.5 & 3.6 are no longer supported. (`#2074 <https://github.com/falconry/falcon/pull/2074>`__)
-- End-of-life Python 3.7 and (soon end-of-life) 3.8 are no longer actively
-  supported, but the framework should still continue to install from source and
-  function.
+- End-of-life Python 3.5 & 3.6 & 3.7 are no longer supported. (`#2074 <https://github.com/falconry/falcon/pull/2074>`__, `#2273 <https://github.com/falconry/falcon/pull/2273>`__)
+- Soon end-of-life Python 3.8 is no longer actively supported, but
+  the framework should still continue to install from source and function.
 - The Falcon 4.x series is guaranteed to support CPython 3.10 and
   PyPy3.10 (v7.3.16).
   This means that we may drop the support for Python 3.7-3.9 altogether in a
```
Comment (vytas7) [inline docs/changes/4.0.0.rst:23]: :arrow_right: `3.8-3.9`

File: docs/changes/4.0.0.rst [hunk 2]
```diff
@@ -15,10 +15,9 @@ Changes to Supported Platforms
 - CPython 3.11 is now fully supported. (`#2072 <https://github.com/falconry/falcon/issues/2072>`__)
 - CPython 3.12 is now fully supported. (`#2196 <https://github.com/falconry/falcon/issues/2196>`__)
 - CPython 3.13 is now fully supported. (`#2258 <https://github.com/falconry/falcon/issues/2258>`__)
-- End-of-life Python 3.5 & 3.6 are no longer supported. (`#2074 <https://github.com/falconry/falcon/pull/2074>`__)
-- End-of-life Python 3.7 and (soon end-of-life) 3.8 are no longer actively
-  supported, but the framework should still continue to install from source and
-  function.
+- End-of-life Python 3.5 & 3.6 & 3.7 are no longer supported. (`#2074 <https://github.com/falconry/falcon/pull/2074>`__, `#2273 <https://github.com/falconry/falcon/pull/2273>`__)
```
Comment (vytas7) [inline docs/changes/4.0.0.rst:18]: `3.5 & 3.6 & 3.7` :arrow_right: `3.5, 3.6 & 3.7`

**Review by vytas7 — CHANGES_REQUESTED** [review]

File: docs/changes/4.0.0.rst (hunk 1 above)
Comment (CaselIT) [reply docs/changes/4.0.0.rst:23]: missed. sorry

**Review by vytas7 — APPROVED** [review]
LGTM now

**Pull request merged**